src/cpp/proxy/domainmap.cpp

```diff
@@ -732,13 +733,13 @@ class DomainMap::Thread : public QThread
 	{
 		worker = new Worker;
 		worker->fileName = fileName;
-		connect(worker, &Worker::started, this, &Thread::worker_started, Qt::DirectConnection);
+		startedConnections[worker] = worker->started.connect(boost::bind(&Thread::worker_started, this));
 		QMetaObject::invokeMethod(worker, "start", Qt::QueuedConnection);
 		exec();
 		delete worker;
```

Comment:
Keeping a map shouldn't be needed here, since the `run()` method runs synchronously from start to finish (it is invoked by `QThread`), and the worker is destroyed at the end.